servicetalk-http-api/src/main/java/io/servicetalk/http/api/DefaultServiceDiscoveryRetryStrategy.java

```diff
@@ -0,0 +1,291 @@
+            return new Builder<>(new IndefiniteRetryStrategy(executor, initialDelay),
+                    evt -> new PartitionedServiceDiscovererEvent<ResolvedAddress>() {
+                        @Override
+                        public PartitionAttributes partitionAddress() {
+                            return evt.partitionAddress();
+                        }
+
+                        @Override
+                        public ResolvedAddress address() {
+                            return evt.address();
+                        }
+
+                        @Override
+                        public boolean isAvailable() {
+                            return !evt.isAvailable();
+                        }
+                    });
+        }
+
+        /**
+         * Creates a new builder that uses default retries.
+         *
+         * @param executor {@link Executor} to use for retry backoffs.
+         * @param initialDelay {@link Duration} to use as initial delay for backoffs.
+         * @param flipAvailability {@link UnaryOperator} that returns a new {@link ServiceDiscovererEvent} that is the
+         * same as the passed {@link ServiceDiscovererEvent} but with {@link ServiceDiscovererEvent#isAvailable()} value
+         * flipped.
+         * @param <ResolvedAddress> The type of address after resolution.
+         * @param <E> Type of {@link ServiceDiscovererEvent}s published from {@link ServiceDiscoverer#discover(Object)}.
+         * @return A new {@link Builder}.
+         */
+        public static <ResolvedAddress, E extends ServiceDiscovererEvent<ResolvedAddress>> Builder<ResolvedAddress, E>
+        withDefaults(final Executor executor, final Duration initialDelay, final UnaryOperator<E> flipAvailability) {
+            return new Builder<>(new IndefiniteRetryStrategy(executor, initialDelay), flipAvailability);
+        }
+    }
+
+    private static final class EventsCache<R, E extends ServiceDiscovererEvent<R>> {
+        @SuppressWarnings("rawtypes")
+        private static final Map NONE_RETAINED = emptyMap();
+
+        private Map<R, E> retainedAddresses = noneRetained();
+        private int targetSize;
+        private final Map<R, E> activeAddresses = new HashMap<>();
+        private final int retainTillReceivePercentage;
+        private final UnaryOperator<E> flipAvailability;
+
+        EventsCache(final int retainTillReceivePercentage, final UnaryOperator<E> flipAvailability) {
+            this.retainTillReceivePercentage = retainTillReceivePercentage;
+            this.flipAvailability = flipAvailability;
+        }
+
+        void errorSeen() {
+            if (retainedAddresses == NONE_RETAINED) {
+                retainedAddresses = new HashMap<>(activeAddresses);
+            } else {
+                retainedAddresses.putAll(activeAddresses);
+            }
+            targetSize = (int) (ceil(retainTillReceivePercentage / 100d) * activeAddresses.size());
```

Comment:
`System.out.println("Output for 0	:" +(int)Math.ceil(0/100d));
		System.out.println("Output for 10	:" +(int)Math.ceil(10/100d));
		System.out.println("Output for 75	:" +(int)Math.ceil(75/100d));
		System.out.println("Output for 100	:" +(int)Math.ceil(100/100d));`

`Output for 0	:0
Output for 10	:1
Output for 75	:1
Output for 100	:1`

Seems like Percentage computation is bit off. Irrespective of percentage targetSize = 1 * activeAddresses.size()

Reply:
Great catch!

Wrong placement of brackets 😢 

I will fix and add tests, thanks!